src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!(herbivore instanceof Carnivore || herbivore.hidden === true)) {
```

Comment:
[CODE QUALITY] The condition in the `if` statement is incorrect. According to the task description, the `bite` method should not work if the object is another carnivore or the herbivore is currently hiding. You should check if the `herbivore` is an instance of `Herbivore` and not hidden instead of checking if it's not an instance of `Carnivore`.